src/main/java/nl/tudelft/ewi/devhub/server/web/resources/repository/AbstractProjectResource.java

```diff
@@ -579,6 +582,46 @@ public Response showSettings(@Context HttpServletRequest request) throws IOExcep
 		return display(templateEngine.process("project/settings.ftl", locales, parameters));
 	}
 
+	@POST
+	@Path("/branches/delete")
+	@Transactional
+	public Response deleteBehindBranch(@Context HttpServletRequest request,
+									   @FormParam("branchDeleteName") String branchDeleteName)
+			throws IOException {
+
+		RepositoryEntity repositoryEntity = getRepositoryEntity();
+		RepositoryApi repositoryApi = repositoriesApi.getRepository(repositoryEntity.getRepositoryName());
+
+        List<Locale> locales = Collections.list(request.getLocales());
+
+        Map<String, Object> parameters = getBaseParameters();
+        parameters.put("repository", repositoryApi.getRepositoryModel());
+
+        if (branchDeleteName != null) {
+            BranchApi branchApi = repositoryApi.getBranch(branchDeleteName);
+
+            try {
+                BranchModel branchModel = branchApi.get();
+
+                if (!branchModel.isAhead()) {
+                    branchApi.deleteBranch();
+                    parameters.put("deleteSuccessful", true);
+                } else {
+                    parameters.put("deleteSuccessful", false);
+                }
+            } catch (NotFoundException ignored) {}
```

Comment:
Oh that is a good one. It didn't always find it here (due to programming error). If it does not find it then it will throw a 404, which will indeed be a better error.